design/spaces.go

```diff
@@ -58,9 +58,10 @@ var spaceOwnedBy = a.Type("SpaceOwnedBy", func() {
 
 var spaceAttributes = a.Type("SpaceAttributes", func() {
 	a.Attribute("name", d.String, "Name for the space", func() {
-		a.MaxLength(62) // maximum space name length is 62 characters
-		a.MinLength(1)  // minimum space name length is 1 characters
-		a.Pattern("^[^_|-].*")
+		a.MaxLength(62)                                             // maximum name length is 62 characters
+		a.MinLength(1)                                              // minimum name length is 1 characters
+		a.Pattern("^(?i)(h(o(m([^e]|$)|[^m]|$)|[^o]|$)|[^h_-]).*$") // Match _|-|home space name
```

Comment:
I will add a comment to the code but:
1) avoid the creation of a space name home (case insensitive)
2) avoid space names that start with _ or -

This was requested if i didn't misunderstand in https://github.com/almighty/almighty-core/pull/1279#issuecomment-300424157